### PR TITLE
Release

### PR DESCRIPTION
**Pull request by HarsimranKaur13**

### CODE REVIEW

File: core_java/grep/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (jarviscanada) [inline core_java/grep/.idea/compiler.xml]: git rm core_java/grep/.idea

File: core_java/grep/data/file1.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+This is some text
+Cat is an animal
```
Comment (jarviscanada) [inline core_java/grep/data/file1.txt]: remove all test files

File: core_java/grep/dependency-reduced-pom.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>ca.jrvs.apps</groupId>
```
Comment (jarviscanada) [inline core_java/grep/dependency-reduced-pom.xml]: remove this

File: core_java/grep/grep.iml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_1_8">
```
Comment (jarviscanada) [inline core_java/grep/grep.iml]: remove this

File: core_java/grep/outFile [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Text us later at night
```
Comment (jarviscanada) [inline core_java/grep/outFile]: remove

File: core_java/grep/target/maven-archiver/pom.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#Generated by Maven
+#Wed Mar 10 19:04:55 UTC 2021
```
Comment (jarviscanada) [inline core_java/grep/target/maven-archiver/pom.properties]: git rm core_java/grep/target

File: linux_sql/linux.jpeg [hunk 7]
```diff
@@ -0,0 +1 @@
...0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36" etag="WbhnyQaaULDpfhsg0b2h" version="14.5.1" type="device"><diagram id="ZYp-jTosQxjIzKAQmC7D" name="Page-1">1ZhRk5owEMc/jY/tGAKij1Wv15m2czdjZ3r25SZCCplDwoQocp++wQSBRKu1KvLisJtkSX75m13Sg5Pl5pGhJPxOfRz1rL6/6cFpz7JcC4rfwpFLB3Qd6QgY8aULVI4ZecfK2VfeFfFx2ujIKY04SZpOj8Yx9njDhxijWbPbbxo135qgABuOmYci0/uT+DyU3qHlVv4vmARh+WYwGMmWJSo7q5WkIfJpVnPBhx6cMEq5fFpuJjgq2JVc5LjPB1p3E2M45qcMeAQZfx5/fR//WL/SLKNPr2+/PqgoaxSt1ILVZHleEmB0Ffu4CAJ6cJyFhONZgryiNRNbLnwhX0aqWYXDjOPNwXmC3eqFajBdYs5y0UUNgJZSiFKMrfhlFX7gKF9YQ+8qH1I7HuwiV1DEg+LyD4ysO2RUQsk1uw7J2gNpcC1I8P4gAdCEZA3bhmTfHyRneG+QHAPSLCPcC/+Cqt/Kn87qt41qcFxPOPY/FWlQWDGNcRMK3hD+UvD76ChrrmgWz9NN3chLIxYzf6kb8ypCYVbDtlY5Ts4M+0a61fCL2dMV8/Dxs4YjFmB+TEnmdta2a18eKX0MR4iTdXO6+7ZQveGZErGQSi0jTS26CuQy1ah63tYDWVogWwskORiBtoraLft8kbn/K7KdYCqNzOtt1xeM1QnBuFpOh2cKxtYFM7ytYIZtnkqgtVPJ7oTI9Bx29qmkH2/gNJGJbUd5rVtSdEivIcNR988t50RJHahbbiQprYIEo3PPLagFunGiKxVcU8wYpaLu7M88RhJuyEd80SfFY8Koh9P0eBG6QN5bsC1bn1Y8IoXiLlacQrsBzzFr0+EeDeg55mKlKTBvFroDE+gZdNQ2TfMOojs0oaMVNq1r07ys6A5NR0+8rWvTvNWYLg4i9HKBw8cMngBRft9/WxyjmspbY+BchrBeP9uOAdjeA/iMr3xhVve/Mo1Vl+jw4Q8=</diagram></mxfile>
```
Comment (jarviscanada) [inline linux_sql/linux.jpeg:1]: fix it

File: linux_sql/scripts/host_info.sh [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+cpu_mhz, l2_cache, total_mem, timestamp) VALUES('$hostname',$cpu_number,'$cpu_architecture','$cpu_model',
+$cpu_mhz, $l2_cache, $total_mem, '$timestamp'),
+('$hostname',2,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 3452, '$timestamp'),
+('$hostname',3,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 4512, '$timestamp'),
+('$hostname',1,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 2165, '$timestamp'),
+('$hostname',2,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 9023, '$timestamp'),
+('$hostname',3,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 4521, '$timestamp'),
+('$hostname',2,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 8734, '$timestamp'),
+('$hostname',3,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 6741, '$timestamp'),
+('$hostname',1,'$cpu_architecture','$cpu_model',$cpu_mhz, $l2_cache, 1254, '$timestamp');"
```
Comment (jarviscanada) [inline linux_sql/scripts/host_info.sh]: remove

File: linux_sql/scripts/ddl.sql [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    cpu_model VARCHAR NOT NULL,
+    cpu_mhz INTEGER NOT NULL,
+    l2_cache INTEGER NOT NULL,
+    total_mem INTEGER NOT NULL,
+    "timestamp" TIMESTAMP NOT NULL
+);
+CREATE TABLE PUBLIC.host_usage
+(
+    "timestamp" TIMESTAMP NOT NULL,
+    host_id SERIAL NOT NULL,
```
Comment (jarviscanada) [inline linux_sql/scripts/ddl.sql:17]: make it a FK

File: linux_sql/scripts/host_usage.sh [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+memory_free=$(cat /proc/meminfo | egrep "^MemFree\:" | awk '{print $2}' | xargs)
+cpu_idle=$(vmstat -t|tail -1|awk '{print $15}')
+cpu_kernel=$(vmstat -t| tail -1|awk '{print $14}')
+disk_io=$(vmstat --unit M | tail -1 | awk '{print $9}')
+disk_available=$(df -BM | awk 'NR==2{print substr($4, 1, length($4)-1)}')
+
+insert_stmt="INSERT INTO host_usage(timestamp,memory_free,cpu_idle,cpu_kernel,disk_io,disk_available)
+VALUES('$timestamp', $memory_free, $cpu_idle, $cpu_kernel, $disk_io, $disk_available);"
+
+export PGPASSWORD=$psql_password
```
Comment (jarviscanada) [inline linux_sql/scripts/host_usage.sh:19]: ID is missing

File: linux_sql/scripts/psql_docker.sh [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+create)
+  line_count='docker container ls -a -f name=jrvs-psql | wc -l'
+  if [ $line_count -eq 2 ] ;
+  then
+  echo 'Created Already'
+  exit 0
+  fi
+
+  if [ -z "$2" ] || [ -z "$3" ] ;
```
Comment (jarviscanada) [inline linux_sql/scripts/psql_docker.sh]: don't use $1 instead, use vars

File: linux_sql/scripts/queries.sql [hunk 12]
```diff
@@ -0,0 +1,12 @@
+SELECT cpu_number,id,total_mem from host_info 
+ORDER BY cpu_number,total_mem DESC;
+
+SELECT AVG(7492108-memory_free),timestamp FROM (SELECT memory_free,date_trunc('hour', timestamp) + date_part('minute', timestamp):: int / 5 * interval '5 min' as timestamp FROM host_usage) as ss
```
Comment (jarviscanada) [inline linux_sql/scripts/queries.sql]: fix it

**Review by jarviscanada — CHANGES_REQUESTED** [review]
Fix all and open another PR

File: linux_sql/scripts/queries.sql (hunk 12 above)
Comment (HarsimranKaur13) [reply linux_sql/scripts/queries.sql]: Done

File: linux_sql/scripts/psql_docker.sh (hunk 11 above)
Comment (HarsimranKaur13) [reply linux_sql/scripts/psql_docker.sh]: Done

File: linux_sql/scripts/host_usage.sh (hunk 10 above)
Comment (HarsimranKaur13) [reply linux_sql/scripts/host_usage.sh:19]: ID type is SERIAL, hence it gets added automatically.

File: linux_sql/scripts/ddl.sql (hunk 9 above)
Comment (HarsimranKaur13) [reply linux_sql/scripts/ddl.sql:17]: Done

File: linux_sql/scripts/host_info.sh (hunk 8 above)
Comment (HarsimranKaur13) [reply linux_sql/scripts/host_info.sh]: Removed

File: core_java/grep/outFile (hunk 5 above)
Comment (HarsimranKaur13) [reply core_java/grep/outFile]: Removed core_java from the develop branch as of now.

File: core_java/grep/grep.iml (hunk 4 above)
Comment (HarsimranKaur13) [reply core_java/grep/grep.iml]: Removed core_java from the develop branch as of now.

File: core_java/grep/dependency-reduced-pom.xml (hunk 3 above)
Comment (HarsimranKaur13) [reply core_java/grep/dependency-reduced-pom.xml]: Removed core_java from the develop branch as of now.

File: core_java/grep/data/file1.txt (hunk 2 above)
Comment (HarsimranKaur13) [reply core_java/grep/data/file1.txt]: Removed core_java from the develop branch as of now.

File: core_java/grep/.idea/compiler.xml (hunk 1 above)
Comment (HarsimranKaur13) [reply core_java/grep/.idea/compiler.xml]: Removed core_java from the develop branch as of now.

File: jarvis_data_eng_Projects.iml [hunk 13]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
+  <component name="NewModuleRootManager" inherit-compiler-output="true">
+    <exclude-output />
```
Comment (jarviscanada) [inline jarvis_data_eng_Projects.iml:4]: remove

**Review by jarviscanada — APPROVED** [review]
please merge

**Pull request merged**